RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -0,0 +1,628 @@
+                   theClusterParam.templXrec_, theClusterParam.templSigmaX_, 
+		   theClusterParam.templProbXY_,
+         	   theClusterParam.probabilityQ_,
+		   theClusterParam.qBin_,
+		   deltay, npixels
+                   );
+   // ******************************************************************
+   
+   //--- Check exit status
+   if unlikely( theClusterParam.ierr2 != 0 )
+   {
+      LogDebug("PixelCPEClusterRepair::localPosition") <<
+      "3D reconstruction failed with error " << theClusterParam.ierr2 << "\n";
+      
+      // Gavril: what do we do in this case ? For now, just return the cluster center of gravity in microns
+      // In the x case, apply a rough Lorentz drift average correction
+      // To do: call PixelCPEGeneric whenever PixelTempReco2D fails
+      float lorentz_drift = -999.9;
+      if ( ! GeomDetEnumerators::isEndcap(theDetParam.thePart) )
+         lorentz_drift = 60.0f; // in microns
+      else
+         lorentz_drift = 10.0f; // in microns
+      // ggiurgiu@jhu.edu, 21/09/2010 : trk angles needed to correct for bows/kinks
+      if ( theClusterParam.with_track_angle )
+      {
+         theClusterParam.templXrec_ = theDetParam.theTopol->localX( theClusterParam.theCluster->x(), theClusterParam.loc_trk_pred ) - lorentz_drift * micronsToCm; // rough Lorentz drift correction
+         theClusterParam.templYrec_ = theDetParam.theTopol->localY( theClusterParam.theCluster->y(), theClusterParam.loc_trk_pred );
+      }
+      else
+      {
+         edm::LogError("PixelCPEClusterRepair")
+         << "@SUB = PixelCPEClusterRepair::localPosition"
+         << "Should never be here. PixelCPEClusterRepair should always be called with track angles. This is a bad error !!! ";
+         
+         theClusterParam.templXrec_ = theDetParam.theTopol->localX( theClusterParam.theCluster->x() ) - lorentz_drift * micronsToCm; // rough Lorentz drift correction
+         theClusterParam.templYrec_ = theDetParam.theTopol->localY( theClusterParam.theCluster->y() );
+      }
+   }   
+   else 
+   {
+      //--- Template Reco succeeded.
+      theClusterParam.hasFilledProb_ = true;
+
+      //--- Go from microns to centimeters
+      theClusterParam.templXrec_ *= micronsToCm;
+      theClusterParam.templYrec_ *= micronsToCm;
+      
+      //--- Go back to the module coordinate system
+      theClusterParam.templXrec_ += lp.x();
+      theClusterParam.templYrec_ += lp.y();
+   }
+   return;
+}
+
+
+
+
+//------------------------------------------------------------------
+//  localError() relies on localPosition() being called FIRST!!!
+//------------------------------------------------------------------
+LocalError
+PixelCPEClusterRepair::localError(DetParam const & theDetParam,  ClusterParam & theClusterParamBase) const
+{
+   
+   ClusterParamTemplate & theClusterParam = static_cast<ClusterParamTemplate &>(theClusterParamBase);
+   
+   //cout << endl;
```

Comment:
Please remove the commented out lines, or replace with some LogDebug or similar

Reply:
OK, if you require this change now.